src/components/modes/teacher/Settings.js

```diff
@@ -79,8 +117,61 @@ class Settings extends Component {
     dispatchCloseSettings();
   };
 
+  handleVerbOnChange = verb => {
+    const {
+      settings: { hiddenVerbs = [] },
+    } = this.props;
+    const checked = !hiddenVerbs.includes(verb);
+    let newHiddenVerbs = [...hiddenVerbs];
+    if (checked) {
+      newHiddenVerbs.push(verb);
+    } else {
+      newHiddenVerbs = newHiddenVerbs.filter(thisVerb => thisVerb !== verb);
+    }
+    this.saveSettings({ hiddenVerbs: newHiddenVerbs });
+  };
+
+  renderActionChecks = () => {
+    const {
+      actions,
+      classes,
+      settings: { hiddenVerbs = [] },
+    } = this.props;
+
+    const verbs = getUniqueVerbs(actions);
```

Comment:
Could we do this in the `mapStateToProps` bit so that we're not constantly sorting on `render`?